src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -59,37 +58,33 @@ public static function analyze(
                     if (!$maybe_type->from_calculation) {
                         self::handleRedundantCast($maybe_type, $statements_analyzer, $stmt);
                     }
+
+                    if ($maybe_type->isSingleStringLiteral()) {
+                        $valid_int_type = new Type\Union([
+                            new Type\Atomic\TLiteralInt((int)$maybe_type->getSingleStringLiteral()->value),
+                        ]);
+                    }
                 }
 
                 $maybe = $maybe_type->getAtomicTypes();
 
                 if (count($maybe) === 1 && current($maybe) instanceof Type\Atomic\TBool) {
-                    $as_int = false;
-                    $type = new Type\Union([
+                    $valid_int_type = new Type\Union([
                         new Type\Atomic\TLiteralInt(0),
                         new Type\Atomic\TLiteralInt(1),
                     ]);
-
-                    if ($statements_analyzer->data_flow_graph instanceof VariableUseGraph
-                    ) {
-                        $type->parent_nodes = $maybe_type->parent_nodes;
-                    }
-
-                    $statements_analyzer->node_data->setType($stmt, $type);
                 }
             }
 
-            if ($as_int) {
-                $type = $valid_int_type ?? Type::getInt();
+            $type = $valid_int_type ?? Type::getInt();
 
-                if ($statements_analyzer->data_flow_graph instanceof VariableUseGraph
-                ) {
-                    $type->parent_nodes = $maybe_type->parent_nodes ?? [];
-                }
-
-                $statements_analyzer->node_data->setType($stmt, $type);
+            if ($statements_analyzer->data_flow_graph instanceof VariableUseGraph
```

Comment:
Please include this block in the conditional `if($maybe_type)`, that way we can remove the `?? []` and it will make more sense

Reply:
This is is impossible, because we need the `$type` variable to be set first.

Reply:
I'm only talking about the conditional with VariableUseGraph. It doesn't make use of `$type`